app/models/miq_action.rb

```diff
@@ -200,6 +200,38 @@ def invoke_action_for_built_in_policy(rec, inputs)
     send(method.to_sym, self, rec, inputs)
   end
 
+  def invoke_or_queue(
+    synchronous,
+    calling_method,
+    role,
+    zone,
+    target,
+    target_method,
+    target_args = [],
+    log_suffix = nil
+  )
```

Comment:
Argument list is too long. Hard to read and use. Consider to pass in `args` hash as one argument (except `:class_name`)